src/webapi/xml_http_request.rs

```diff
@@ -0,0 +1,69 @@
+use webcore::value::{
+    Reference,
+    Value,
+};
+use webcore::try_from::TryInto;
+
+/// XMLHttpRequest object to make http requests
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest)
+#[derive(Debug)]
+pub struct XMLHttpRequest {
+    reference: Reference,
+}
+
+impl XMLHttpRequest {
+    /// Creates an empty XMLHttpRequest
+    pub fn new() -> XMLHttpRequest {
+        let reference: Reference = js! {
+            var xmlHttp = new XMLHttpRequest();
+            return xmlHttp;
+        }.try_into().unwrap();
+        XMLHttpRequest {
+            reference,
+        }
+    }
+    /// Sets the callback to call once a request completes
+    /// First argument is the status code, the second argument is the response text
+    /// Makes use of onreadystatechange:
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/onreadystatechange)
+    pub fn set_callback<F:FnMut(u32, String) + 'static>(&self, callback: F) {
+        let reference = self.reference.as_ref();
+        js! {
+            var callback = @{callback};
+            var xmlHttp = @{reference};
+            xmlHttp.onreadystatechange = function() {
+                if (xmlHttp.readyState == 4) {
+                    callback(xmlHttp.status, xmlHttp.responseText);
+                }
+            }
+        }
+    }
+
+    /// Open connection with given method (ie GET or POST), and the url to hit
```

Comment:
For every method you expose please take the docs from MDN and lightly edit them as needed, just as in your other PR. (: